src/ResourceManager/DeploymentManager/Commands.DeploymentManager/Commands/NewServiceUnit.cs

```diff
@@ -0,0 +1,138 @@
+    public class NewServiceUnit : DeploymentManagerBaseCmdlet
+    {
+        private const string IncrementalDeploymentMode = "Incremental";
+
+        private const string CompleteDeploymentMode = "Complete";
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "The resource group.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "The name of the serivce topology this service unit is a part of.")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceTopologyName { get; set; }
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "The name of the service this service unit is a part of.")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "The name of the service unit.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "The location of the service unit resource.")]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "Determines the location where resources under the service unit would be deployed to.")]
+        [ValidateNotNullOrEmpty]
+        public string TargetResourceGroup { get; set; }
+
+        [Parameter(
+            Mandatory = true, 
+            HelpMessage = "The deployment mode to use when deploying the resources in the service unit.")]
+        [ValidateSet(NewServiceUnit.IncrementalDeploymentMode, NewServiceUnit.CompleteDeploymentMode)]
+        [ValidateNotNullOrEmpty]
+        public string DeploymentMode { get; set; }
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "The deployment mode to use when deploying the resources in the service unit.")]
+        public string ParametersUri { get; set; }
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "The deployment mode to use when deploying the resources in the service unit.")]
+        public string TemplateUri { get; set; }
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "The deployment mode to use when deploying the resources in the service unit.")]
+        public string TemplateArtifactSourceRelativePath { get; set; }
+
+        [Parameter(
+            Mandatory = false, 
+            HelpMessage = "The deployment mode to use when deploying the resources in the service unit.")]
+        public string ParametersArtifactSourceRelativePath { get; set; }
```

Comment:
the descriptions here need to be updated.. all params have the same description. Also how is this and the one param above it related?

Reply:
Updated the description on all of these. The param 'Uri' is to be used when users are specifying a full URI to the file. The relative path is to be used when the users have already referenced an artifact source resourceId in the parent Topology resource.